saleor/payment/gateways/np_atobarai/api_types.py

```diff
@@ -0,0 +1,70 @@
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Dict, List, NamedTuple
+
+from .const import (
+    FILL_MISSING_ADDRESS,
+    MERCHANT_CODE,
+    NP_TEST_URL,
+    NP_URL,
+    SHIPPING_COMPANY,
+    SKU_AS_NAME,
+    SP_CODE,
+    TERMINAL_ID,
+    USE_SANDBOX,
+)
+
+
+class NPResponse(NamedTuple):
+    result: dict
+    error_codes: List[str]
+
+
+def error_np_response(error_message: str) -> NPResponse:
+    return NPResponse({}, [error_message])
+
+
+@dataclass
+class ApiConfig:
+    url: str
+    fill_missing_address: bool
+    merchant_code: str
+    terminal_id: str
+    sp_code: str
+    shipping_company: str
+    sku_as_name: bool
+
+
+class PaymentStatus(str, Enum):
+    SUCCESS = "00"
+    PENDING = "10"
+    FAILED = "20"
+
+
+@dataclass
+class PaymentResult:
+    status: PaymentStatus
+    psp_reference: str = ""
```

Comment:
I think that this is incorrect. If we don't always have `psp_reference` then I think this field should be marked as `Optional[str]` as it is more meaningful that setting empty `str` as a default one

Reply:
Resolved in recent commit.